modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/data/provider/ContentDashboardDataProvider.java

```diff
@@ -27,117 +31,190 @@
+				assetVocabularies.get(0), assetVocabularies.get(1)));
+	}
 
-		if (!ArrayUtil.isEmpty(audienceAssetCategoryIds)) {
-			termsAggregation = _getTermsAggregation(audienceAssetCategoryIds);
+	private Map<String, String> _getAssetCategoryTitlesMap(
+		AssetVocabulary assetVocabulary, Locale locale) {
 
-			if (!ArrayUtil.isEmpty(stageAssetCategoryIds)) {
-				termsAggregation.addChildAggregation(
-					_getTermsAggregation(stageAssetCategoryIds));
-			}
-		}
-		else {
-			termsAggregation = _getTermsAggregation(stageAssetCategoryIds);
-		}
+		List<AssetCategory> categories = assetVocabulary.getCategories();
 
-		SearchRequestBuilder searchRequestBuilder =
-			_searchRequestBuilderFactory.builder(_searchContext);
-
-		return _toAssetCategoryMetrics(
-			_searcher.search(
-				searchRequestBuilder.addAggregation(
-					termsAggregation
-				).emptySearchEnabled(
-					true
-				).highlightEnabled(
-					false
-				).size(
-					0
-				).build()));
+		Stream<AssetCategory> stream = categories.stream();
+
+		return stream.collect(
+			Collectors.toMap(
+				entry -> String.valueOf(entry.getCategoryId()),
+				entry -> entry.getTitle(locale)));
 	}
 
-	private TermsAggregation _getTermsAggregation(String[] assetCategoryIds) {
-		TermsAggregation termsAggregation = _aggregations.terms(
-			"categories", "assetCategoryIds");
+	private AssetVocabularyMetric _getAssetVocabularyMetric(
+		AssetVocabulary assetVocabulary) {
 
-		termsAggregation.setIncludeExcludeClause(
-			new IncludeExcludeClauseImpl(assetCategoryIds, new String[0]));
+		Map<String, String> assetCategoryTitlesMap = _getAssetCategoryTitlesMap(
+			assetVocabulary, _locale);
 
-		return termsAggregation;
+		return _toAssetVocabularyMetric(
+			assetCategoryTitlesMap, assetVocabulary,
+			_getBuckets(_getTermsAggregation(assetCategoryTitlesMap.keySet())));
 	}
 
-	private List<AssetCategoryMetric> _toAssetCategoryMetrics(
-		SearchResponse searchResponse) {
+	private AssetVocabularyMetric _getAssetVocabularyMetric(
+		AssetVocabulary assetVocabulary, AssetVocabulary childAssetVocabulary) {
+
+		Map<String, String> assetCategoryTitlesMap = _getAssetCategoryTitlesMap(
+			assetVocabulary, _locale);
+
+		Map<String, String> childAssetCategoryTitlesMap =
+			_getAssetCategoryTitlesMap(childAssetVocabulary, _locale);
+
+		TermsAggregation termsAggregation = _getTermsAggregation(
+			assetCategoryTitlesMap.keySet());
 
-		List<AssetCategoryMetric> assetCategoryMetrics = new ArrayList<>();
+		termsAggregation.addChildAggregation(
+			_getTermsAggregation(childAssetCategoryTitlesMap.keySet()));
+
+		return new AssetVocabularyMetric(
+			String.valueOf(assetVocabulary.getVocabularyId()),
+			assetVocabulary.getTitle(_locale),
+			_toAssetCategoryMetrics(
+				assetCategoryTitlesMap, _getBuckets(termsAggregation),
+				childAssetCategoryTitlesMap, childAssetVocabulary));
+	}
+
+	private Collection<Bucket> _getBuckets(TermsAggregation termsAggregation) {
+		Collection<String> classNames =
```

Comment:
We can't duplicate logic from the Searcher and the Aggregations, this could generate some future bugs, we should find a solution which fits the search and the Aggregations, so when we change the Search of the Aggregations, the data keeps working the same way.